doc/contribute/guidelines.rst

```diff
@@ -554,6 +570,38 @@ exceptions:
 * Avoid using binary literals (constants starting with ``0b``).
 * Avoid using non-ASCII symbols in code, unless it significantly improves
   clarity, avoid emojis in any case.
+* Use proper capitalization of nouns in code comments (e.g. ``UART`` and not
+  ``uart``, ``CMake`` and not ``cmake``).
+
+Beyond C code, the following coding style rules apply to other types of files:
+
+* CMake
+
+  * Indent with spaces, indentation is two spaces.
+  * No space between commands (e.g. ``if``) and the corresponding opening
+    bracket (e.g. ``(``).
+
+* Devicetree
+
+  * Indent with tabs.
+  * Follow the Devicetree specification conventions and rules.
+  * Use dashes (``-``) as word separators for node and property names.
+  * Use underscores (``_``) as word separators in node labels.
```

Comment:
The term is `nodelabel` not `node label`.

Reply:
@keith-zephyr I could not find the term `nodelabel` in the spec, nor in our doc except for an example. Am I missing something?

Reply:
it used to be a fairly big source of confusion when label properties were common, probably not so much these days though I see more of those coming back for some reason I don't quite understand

Reply:
Regardless, those are actually named labels in the spec, and in fact they apply both to nodes and properties:

> The source format allows labels to be attached to any node or property value in the devicetree

Reply:
right so it sounds like they are interchangeable, which they are not

Reply:
I see our documentation refers to the term as *node label*.

For new users, *label* and *node label* are quite easy to confuse.  For consistency with all the DT macros, I do think it's worthwhile to explicitly define *nodelabel` (no space).  But that's outside the scope of this PR.

Reply:
Made a note, will send a separate PR.